sdk/src/main/java/com/google/cloud/dataflow/sdk/io/CompressedSource.java

```diff
@@ -424,7 +455,10 @@ protected final boolean readNextRecord() throws IOException {
      */
     @Override
     protected final long getCurrentOffset() {
-      return readerDelegate.getCurrentOffset();
+      if (splittable) {
```

Comment:
Why this change? getCurrentOffset() is documented to return offset of the current record, rather than the last split point before it.